plugins/org.polarsys.capella.scenario.editor/src/org/polarsys/capella/scenario/editor/helper/EmbeddedEditorInstanceHelper.java

```diff
@@ -395,4 +396,24 @@ public static String getModelContent() {
     return ScenarioExt.getAvailableFunctionsStateFragment(element.getRepresentedInstance()).stream()
         .collect(Collectors.toList());
   }
+  
+  public static String getStateFragmentType(StateFragment capellaStateFragment) {
+    if (capellaStateFragment.getRelatedAbstractFunction() != null) {
+      return DslConstants.FUNCTION;
+    } else if (capellaStateFragment.getRelatedAbstractState() != null
+        && capellaStateFragment.getRelatedAbstractState() instanceof State) {
+      return DslConstants.STATE;
```

Comment:
I remember that Mode extends State. in this case will return "state", even if it's mode. you could check if is instanceof Mode and return DslConstants.MODE and else return state 
also you could remove all 'else'-s, they are not necessary

Reply:
Done.